bcda/api_test.go

```diff
@@ -31,14 +39,225 @@ func (s *APITestSuite) SetupTest() {
 	s.rr = httptest.NewRecorder()
 }
 
+func (s *APITestSuite) TestBulkRequest() {
+	s.SetupAuthBackend()
+
+	acoID := "0c527d2e-2e8a-4808-b11d-0fa06baf8254"
+	user, err := s.AuthBackend.CreateUser("api.go Test User", "testbulkrequest@example.com", uuid.Parse(acoID))
+	if err != nil {
+		s.T().Error(err)
+	}
+
+	token := jwt.New(jwt.SigningMethodRS512)
+	token.Claims = jwt.MapClaims{
+		"sub": user.UUID.String(),
+		"aco": acoID,
+		"id":  uuid.NewRandom().String(),
+	}
+	token.Valid = true
+
+	req := httptest.NewRequest("GET", "/api/v1/Patient/$export", nil)
+	req = req.WithContext(context.WithValue(req.Context(), "token", token))
+
+	queueDatabaseURL := os.Getenv("QUEUE_DATABASE_URL")
+	pgxcfg, err := pgx.ParseURI(queueDatabaseURL)
+	if err != nil {
+		s.T().Error(err)
+	}
+
+	pgxpool, err := pgx.NewConnPool(pgx.ConnPoolConfig{
+		ConnConfig:   pgxcfg,
+		AfterConnect: que.PrepareStatements,
+	})
+	if err != nil {
+		s.T().Error(err)
+	}
+	defer pgxpool.Close()
+
+	qc = que.NewClient(pgxpool)
+
+	handler := http.HandlerFunc(bulkRequest)
+	handler.ServeHTTP(s.rr, req)
+
+	assert.Equal(s.T(), http.StatusAccepted, s.rr.Code)
+	s.db.Where("user_id = ?", user.UUID).Delete(models.Job{})
+	s.db.Where("uuid = ?", user.UUID).Delete(auth.User{})
```

Comment:
why delete these?  We should be able to continue with these rows in the DB.

Reply:
Just for the sake of keeping the test database clean-- having extra data could impact other tests

Reply:
I understand.  I just feel like if randomly created data does impact other tests we should figure that out now because we will have multiple users and jobs when this goes live and if things are colliding that will be really bad.

Reply:
Standard test practices are that test cases should execute independently.  We do not want to go down a path where test cases executed after this one start relying on test data that was created as part of this test case.  Test suites (or even better, test cases) should be responsible for setting up and tearing down their own data.
Knoll, for your concern regarding test data affecting multiple users/jobs... this is a use case for integration testing.  For now, all we care about in the unit test exercise is whether or not the module/function is working properly.

Reply:
I get that future tests should not rely on this data and I am not trying to make them do so.  However, we will almost certainly NOT be deleting job records in production. If this was data that would regularly be pruned deleting makes sense but we will likely never delete job records for any reason. Therefore, it is important that having multiple unrelated jobs in the DB have no impact or else some other unit is not working as expected in a way that our testing process ignores by design.  With our focus on unit test coverage and cleaning data we will miss problems of a specific type.  We aren't currently doing any integration testing and it isn't on the PI7 board in any meaningful way so I don't think relying on them is feasible.